### PR TITLE
Joint kernel density plot

### PR DESCRIPTION
**Pull request by fredmontet**
Here's the KDE plot. 

Also, I added some improvements to :

- status plot
- prediction plot
- resolution function in time series

as well as some bugfix to :

- Prophet model

### CODE REVIEW

File: src/timeatlas/plots/time_series.py [hunk 1]
```diff
@@ -1,59 +1,107 @@
...
+        if "unit" in ts.metadata:
+            unit = ts.metadata["unit"]
+            ax.set_ylabel("{} $[{}]$".format(unit.name, unit.symbol))
         if "sensor" in ts.metadata:
             sensor = ts.metadata["sensor"]
-            duration = str(pred.duration())
+            ax.set_title("{}—{}".format(sensor.id, sensor.name))
+
+
+def prediction(forecast: 'TimeSeries', observation: 'TimeSeries' = None):
```
Comment (RychenerLorenz) [inline src/timeatlas/plots/time_series.py:49]: return type "-> NoReturn"

File: src/timeatlas/plots/time_series.py [hunk 2]
```diff
@@ -1,59 +1,107 @@
...
+        ax.plot(observation.series.index, observation.series.values,
+                ls='-',
+                c="0.3",
+                label="observation")
+
     ax.legend()
 
 
-def status(ts: TimeSeries):
+def status(ts: 'TimeSeries', cmap: str = "autumn_r"):
```
Comment (RychenerLorenz) [inline src/timeatlas/plots/time_series.py:97]: add return type "->NoReturn"

File: src/timeatlas/plots/time_series.py [hunk 3]
```diff
@@ -70,17 +118,52 @@ def status(ts: TimeSeries):
...
 
     ax = add_metadata_to_plot(ts.metadata, ax)
 
     fig.autofmt_xdate()
     plt.grid(b=True, which='both')
     plt.colorbar(m, aspect=5, pad=0.01)
     plt.show()
+
+
+def kde(ts: 'TimeSeries'):
```
Comment (RychenerLorenz) [inline src/timeatlas/plots/time_series.py:136]: add return type

File: src/timeatlas/time_series/time_series.py [hunk 4]
```diff
@@ -112,6 +116,29 @@ def create(start: str, end: str, freq: Union[str, 'TimeSeries'] = None,
                            index=date_range(start, end, freq=freq))
         return TimeSeries(series, metadata)
 
+    def register_plotting_function(self, plotting_function: Callable):
```
Comment (RychenerLorenz) [inline src/timeatlas/time_series/time_series.py:119]: add return type

File: src/timeatlas/plots/time_series.py (hunk 1 above)
Comment (fredmontet) [reply src/timeatlas/plots/time_series.py]: Done with another return type

File: src/timeatlas/plots/time_series.py (hunk 2 above)
Comment (fredmontet) [reply src/timeatlas/plots/time_series.py]: Done with another return type

File: src/timeatlas/plots/time_series.py (hunk 3 above)
Comment (fredmontet) [reply src/timeatlas/plots/time_series.py]: Done with another return type

File: src/timeatlas/time_series/time_series.py (hunk 4 above)
Comment (fredmontet) [reply src/timeatlas/time_series/time_series.py]: Done

**Comment by fredmontet** [comment]
Et voilà!

**Pull request merged**